play/src/main/scala/JwtSession.scala

```diff
@@ -124,7 +131,7 @@ object JwtSession extends JwtJsonImplicits with JwtPlayImplicits {
     case _ => Json.obj()
   }
 
-  def defaultHeader(implicit conf: Configuration): JwtHeader = key.map(_ => JwtHeader(ALGORITHM)).getOrElse(JwtHeader())
+  def defaultHeader(implicit conf: Configuration): JwtHeader = secretKey.map(_ => JwtHeader(ALGORITHM)).getOrElse(JwtHeader())
```

Comment:
Shouldn't it be `privateKey`? The goal is to put the algorithm in the header if there is a key defined.

Reply:
It depends on selected algorithm. Fixed this part.